webserver/views/user.py

```diff
@@ -98,6 +99,63 @@ def import_data():
                            lastfm_username=lastfm_username)
 
 
+@user_bp.route("/upload", methods=['POST'])
+@login_required
+def upload():
+    if request.method == 'POST':
+        f = request.files['file']
+        if f:
+            augmented_listens = []
+            msb_listens = []
+            try:
```

Comment:
The data file that I sent you isn't actually the whole archive file. It is in fact a zip, containing many csvs and many json files. I've attached an example.
[21192540-QD3jC21IfeW40cZFN7AJ5AY7upcq49FR.zip](https://github.com/metabrainz/listenbrainz-server/files/204213/21192540-QD3jC21IfeW40cZFN7AJ5AY7upcq49FR.zip)
We should import the json data.